src/handlers/handleIngressAPI.ts

```diff
@@ -1,13 +1,28 @@
-import { IntegrationEnv } from '../env'
+import {
+  FingerprintDecryptionKeyName,
+  FingerprintSecretStoreName,
+  IntegrationEnv,
+  isOpenClientResponseEnabled,
+} from '../env'
 import {
   addProxyIntegrationHeaders,
   addTrafficMonitoringSearchParamsForVisitorIdRequest,
   createErrorResponseForIngress,
   createFallbackErrorResponse,
 } from '../utils'
 import { getFilteredCookies } from '../utils/cookie'
+import { SecretStore } from 'fastly:secret-store'
+import { unsealData } from '../utils/unsealData'
+import { processUnsealedResult } from '../utils/processUnsealedResult'
 
 async function makeIngressRequest(receivedRequest: Request, env: IntegrationEnv) {
+  // Get decryption key from secret store
+  const secretStore = new SecretStore(FingerprintSecretStoreName)
+  const decryptionKey = await secretStore.get(FingerprintDecryptionKeyName).then((v) => v?.plaintext())
+  if (!decryptionKey) {
+    throw new Error('Decryption key not found in secret store')
```

Comment:
This whole `decryptionKey` logic needs to be under the flag of `isOpenClientResponseEnabled` no? If I don't have unsealing result feature enabled, this would throw an error for me.

Reply:
Thank you, moved this!